### DISCUSSION TITLE
zeroc-ice static build library name wrong?

### DISCUSSION THREAD

**Discussion by jvbsl** [microsoft/vcpkg#33575] [Q&A]
Hello there,

I have a little problem with zeroc-ice:x64-windows-static-md, as cmake does not find the library ice37++11.lib.
Which in some way is correct, because it does not exist, only ice++11.lib exists. With the x64-windows(shared) build the lib name is indeed ice37++11.lib.

So now my first question would be which one is the "correct" way for vcpkg library names? Because as far as I can see some packages have the versions in them, some do not. Do we expect a difference between static and shared builds?

Now I know where it stems from in the zeroc-ice build:
If we use cmake to build zeroc-ice the shared and static build are both named the same: ice37++11.lib
If we use vcxproj(used by the vcpkg port) the shared build is named ice37++11.lib and the static build ice++11.lib

I could not track down where the naming scheme inside vcxproj stems from, is that perhaps a Microsoft standard and done by msbuild automatically without it being set anywhere explicitly?
Does anyone know how to workaround that?

Thank you so much in advance for all the help and info you can give me

**Comment by FrankXie05** [comment]
@jvbsl  You should create an issue for this.

**Comment by jvbsl** [comment]
Yeah I wasn't sure if that would be fitting for an issue because it contains generic vcpkg questions, but in that case I'll do it, thanks
https://github.com/microsoft/vcpkg/issues/33589
